### PR TITLE
PHP: add unit test to increase PHP C extension code coverage

### PR DESCRIPTION
**Pull request by stanley-cheung**
For #3780 

The PHP C extension code coverage is now 95.1%

https://grpc-testing.appspot.com/job/gRPC_pull_requests/6229/config=gcov,language=php,platform=linux/HTML_Report/

### CODE REVIEW

File: src/php/ext/grpc/call.c [hunk 1]
```diff
@@ -125,15 +125,9 @@ zval *grpc_parse_metadata_array(grpc_metadata_array *metadata_array) {
     memcpy(str_val, elem->value, elem->value_length);
     if (zend_hash_find(array_hash, str_key, key_len, (void **)data) ==
         SUCCESS) {
-      if (Z_TYPE_P(*data) != IS_ARRAY) {
-        zend_throw_exception(zend_exception_get_default(),
-                             "Metadata hash somehow contains wrong types.",
-                             1 TSRMLS_CC);
-          efree(str_key);
-          efree(str_val);
```
Comment (murgatroid99) [inline src/php/ext/grpc/call.c]: Why are we removing frees here?

Comment (stanley-cheung) [inline src/php/ext/grpc/call.c]: I find that this is pretty much unreachable. All the cases where `grpc_parse_metadata_array` is called, the metadata has already been cleaned/verified by a previous call.

Comment (murgatroid99) [inline src/php/ext/grpc/call.c]: Actually, I just realized that you removed a whole conditional here. That change should make this function unconditionally fail on any non-empty array.

Comment (murgatroid99) [inline src/php/ext/grpc/call.c]: And if the condition is unreachable, just remove it. Otherwise, handle it properly. But don't just remove part of the handling code.

Comment (stanley-cheung) [inline src/php/ext/grpc/call.c]: Actually let me just revert this change, probably unnecessary.

Comment (stanley-cheung) [inline src/php/ext/grpc/call.c]: Reverted

File: src/php/ext/grpc/channel.c [hunk 2]
```diff
@@ -148,37 +148,33 @@ PHP_METHOD(Channel, __construct) {
                          "Channel expects a string and an array", 1 TSRMLS_CC);
     return;
   }
-  if (args_array == NULL) {
```
Comment (murgatroid99) [inline src/php/ext/grpc/channel.c]: According to the code a few lines up, `args_array` is an optional argument, so it can be `NULL`. Please either change that or revert this.

Comment (stanley-cheung) [inline src/php/ext/grpc/channel.c]: Yes all caller to the `Channel` constructor - the array is required. Updated the few lines up. The null case should be unreachable.

**Comment by stanley-cheung** [comment]
Thanks for the review.
